### ISSUE TITLE
Facial Emotion Detection

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#232]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Facial Emotion Detection</br>
:red_circle: **Aim** : Create a DL model which will detect the emotions from the facial activities.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/mhantor/facial-expression</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by KSatwik** [comment]
please assign this task to me— Facial Emotion Detection
FullName: Satwik Reddy Katukuri
Github Profile Link: https://github.com/KSatwik
Email ID: satwik.vnr@gmail.com

Approach for this Project:
Load and preprocess the facial expression dataset.
Split the dataset into training and testing sets.
Choose 3-4 deep learning algorithms suitable for image classification, such as CNN, RNN, or their combinations.
Implement each selected model separately.
Train each model using the training dataset.
Adjust hyperparameters for optimal performance.
Evaluate the trained models using the testing dataset.
Compare the performance of different models based on accuracy.
Select the model with the highest accuracy as the best-fitted algorithm.

What is your participant role? -contributor at Social Summer of Code (SSOC), Season-2

**Comment by abhisheks008** [comment]
Hi @KSatwik  nice to have you here. I like your approach for solving this issue. This issue will be assigned to you once the program starts officially.

Till then, star the repository and explore more in open source.

**Comment by abhisheks008** [comment]
Issue assigned to @KSatwik

**Comment by siddhant4ds** [comment]
@abhisheks008 Is this issue available for SWOC-4? If yes, please assign it to me.

* Full name : Siddhant Tiwari
* GitHub Profile Link : https://github.com/siddhant4ds
* Email ID : siddhant.tiwari.ds@gmail.com
* Participant ID (if applicable): N/A
* Approach for this Project : 
   1. Baseline model using a simple CNN
   2. Explore techniques for improving image quality (since they are very low-resolution)
   3. Transfer-learning using different architectures as feature-extraction backbone
   4. Fine-tuning the best performing transfer learning model

* What is your participant role? (Mention the Open Source program) SWOC-4 (2024)

**Comment by abhisheks008** [comment]
_**Suggestion**_: Try to use atleast 2-3 different Deep learning techniques/methods for this project and compare them using the accuracy scores to find out the best fitted model for this dataset.

Issue assigned to you @siddhant4ds
